### PR TITLE
Fix SQL Statements

### PR DESCRIPTION
**Pull request by Narimm**
This moves SQL to use try with resources rather than long winded null check closure.

Signed-off-by: Narimm <benjicharlton@gmail.com>

### CODE REVIEW

**Comment by viveleroi** [comment]
I sure wish try-with-resources was a thing when I started Prism and all my other db-using mods. I used this in the sponge Prism and it's so much cleaner.

File: src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java [hunk 1]
```diff
@@ -116,35 +132,36 @@ public void handleDataSourceException(SQLException e) {
...
         } catch (final SQLException ignored) {
+            log.error("Database rescue was unsuccessful.");
         }
         log.error("Database connection error: " + e.getMessage());
         if (e.getMessage().contains("marked as crashed")) {
             final String[] msg = new String[2];
             msg[0] = "If MySQL crashes during write it may corrupt it's indexes.";
-            msg[1] = "Try running `CHECK TABLE " + getPrefix() + "data` and then `REPAIR TABLE " + getPrefix() + "data`.";
+            msg[1] = "Try running `CHECK TABLE " + getPrefix() + "data` and then `REPAIR TABLE "
+                    + getPrefix() + "data`.";
```
Comment (alchemistmatt) [inline src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java:142]: I'd remove the period, changing to 
"data`";

File: src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java [hunk 2]
```diff
@@ -116,35 +132,36 @@ public void handleDataSourceException(SQLException e) {
...
-        Statement st = null;
-        try {
-            conn = getConnection();
-            if (conn == null)
-                return;
-
-            // actions
-            String query = "CREATE TABLE IF NOT EXISTS `" + getPrefix() + "actions` ("
+        try (
+                Connection  conn = getConnection();
```
Comment (alchemistmatt) [inline src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java:154]: minor point, but you have two spaces here: `Connection  conn`

File: src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java [hunk 3]
```diff
@@ -207,81 +224,57 @@ public void setupDatabase(ActionRegistry actionRegistry) {
...
-        try {
-            conn = getConnection();
-            s = conn.prepareStatement("SELECT action_id, action FROM " + getPrefix() + "actions");
-            rs = s.executeQuery();
 
+        try (
+                Connection conn = getConnection();
+                PreparedStatement s = conn.prepareStatement(
+                        "SELECT action_id, action FROM " + prefix + "actions");
+                ResultSet rs = s.executeQuery()
```
Comment (alchemistmatt) [inline src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java:276]: Does `ResultSet rs = s.executeQuery()` need to end with a semicolon?  If it does, compilation should fail ... and I'm sure you're compiled before committing, :wink:

File: src/main/java/me/botsko/prism/players/PlayerIdentification.java [hunk 4]
```diff
@@ -1,482 +1,360 @@
...
+
+        if (player != null) {
+            return getPrismPlayer(player);
+        }
+
+        // Player not online, we need to go to cache
+
+        return lookupByName(playerName);
+
+        // No player exists! We must create one
```
Comment (alchemistmatt) [inline src/main/java/me/botsko/prism/players/PlayerIdentification.java]: This comment should be removed

File: src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java (hunk 3 above)
Comment (alchemistmatt) [reply src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java:276]: Oh !  I see; this is like a `using` statement in C#, but instead a localized try/catch that exists during the scope of `rs`

File: src/main/java/me/botsko/prism/players/PlayerIdentification.java [hunk 5]
```diff
@@ -1,482 +1,360 @@
...
+        }
+        return fakePlayer;
+    }
+
+    /**
+     * Saves a player's UUID to the prism_players table. We cache the current
+     * username as well.
+     */
+    protected static void updatePlayer(PrismPlayer prismPlayer) {
+        prefix = Prism.config.getString("prism.mysql.prefix","prism_");
```
Comment (alchemistmatt) [inline src/main/java/me/botsko/prism/players/PlayerIdentification.java:264]: Great addition of a default value of `prism_` if the config doesn't have `prism.mysql.prefix` defined.

**Review by alchemistmatt — COMMENTED** [review]
See code comments

**Review by alchemistmatt — DISMISSED** [review]

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 13
- Added 1
           

Complexity decreasing per file
==============================
+ src/main/java/me/botsko/prism/players/PlayerIdentification.java  -4
+ src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java  -6
         

Clones removed
==============
+ src/main/java/me/botsko/prism/players/PlayerIdentification.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5441397&bid=17843931)

File: src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java [hunk 6]
```diff
@@ -24,6 +24,7 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
+import java.util.HashMap;
```
Comment (Narimm) [inline src/main/java/me/botsko/prism/database/sql/SQLPrismDataSource.java:27]: Codacy Issue found: [Avoid unused imports such as 'java.util.HashMap'](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5441397)

**Pull request merged**